syntax = "proto3";

// protoc *.proto --python_out=.

package lieferbot;

// External request sent per order update
message OrderUpdate {
    // The id of the order
    string id = 1;
    // The vehicle id of the order
    string vehicle = 2;
    // The current status of the order
    string status = 3;
    // The time the update was made
    float time = 4;
}

// An internal message used to store state of the order
message OrderState {
    // The current state of the order
    string status = 1;
}

// An internal message used to store timestamps
message Time {
    float time = 1;
}

// A customized monitoring report of an order
message Report {
    string id = 1;
    string vehicle = 2;
    float timeUnassigned = 3;
    float timeAssigned = 4;
    float timeInProgress = 5;
    float timeDelivered = 6;
}
